docs/devguide/debugging_guide.rst

```diff
@@ -0,0 +1,70 @@
+
+You should also see the DHCP proxy return the bootfile. In the DHCP-proxy logs, look for lines with `DHCP.messageHandler`::
+
+    S 2016-01-08T19:31:43.268Z [on-dhcp-proxy] [DHCP.messageHandler] [Server] Unknown node 08:00:27:f3:9f:2e. Sending down default bootfile.
+
+And immediately thereafter, you should see the server request the file from TFTP::
+
+    S 2016-01-08T19:31:43.352Z [on-tftp] [Tftp.Server] [Server] tftp: 67.300 monorail.ipxe
+
+.. _WebSequenceDiagrams: https://www.websequencediagrams.com
+
+
+Default discovery workflow
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code::
+
+    title Default Discovery Workflow
+    Server->RackHD: DHCP from PXE(nic or BIOS)
+    RackHD->Server: ISC DHCP response with IP
+    RackHD->Server: DHCP-proxy response with bootfile
+    note right of RackHD: If the node is already "known", it will only respond if there's an active workflow that's been invoked related to the node
+    Server->RackHD: Request to download bootfile via TFTP
+    RackHD->Server: TFTP sends requested file (monorail.ipxe)
+    note left of Server: Server loads monorail.ipxe and initiates on bootloader
+    Server->RackHD: IPXE script requests what to do from RackHD (http)
+    note right of RackHD: RackHD looks up IP address of HTTP request from IPXE script to find the node via it's mac-address.
+    note right of RackHD: If the node is already "known", it will only respond if there's an active workflow that's been invoked related to the node
+    note right of RackHD: If the node isn't known, it will create a workflow (default is a discovery workflow) and respond with an iPXE script to initiate that
+    RackHD->Server: ipxe script (what RackHD calls a Profile) (via http)
+    note left of Server: ipxe script with discovery microkernel and initrd (http)
+    Server->RackHD: ipxe requests static file - the vmlinuz kernel
+    RackHD->Server: vmlinuz (http)
+    Server->RackHD: ipxe requests static file - initrd
+    RackHD->Server: initrd (http)
+    note left of Server: Server loads the kernel and initrd and transfers control (boots that microkernel)
+    Server->RackHD: initrd loads additional file (overlay) from Server to extend microkernel
+    note left of Server: the discovery microkernel is set to request and launch a NodeJS task runnner
+    Server->RackHD: requests the bootstrap.js template
+    RackHD->Server: bootstrap.js filled out with values specific to the node based on a lookup
+    note left of Server: runs node bootstrap.js
+    Server->RackHD: bootstrap asks for tasks (what should I do?)
+    RackHD->Server: data packet of tasks (via http)
+    note left of Server: Discovery Workflow passes down tasks to interrogate hardware
+    loop for each Task from RackHD
+        Server->RackHD: output of task
+    end
+    note right of RackHD: task output stored as catalogs in RackHD related to the node
+    note right of RackHD: if RackHD is configured with SKU definitions, it processes these catalogs to determine the SKU
```

Comment:
After SKU is determined, RackHD will continue do following work [discovery-sku-graph.js](https://github.com/RackHD/on-taskgraph/blob/master/lib/graphs/discovery-sku-graph.js):
- Generate enclosure nodes bases on catalog data.
- Create default pollers.